src/ado2gh/Program.cs

```diff
@@ -39,6 +40,7 @@ public static async Task Main(string[] args)
                 .AddSingleton<AdoInspectorServiceFactory>()
                 .AddSingleton<DateTimeProvider>()
                 .AddSingleton<FileSystemProvider>()
+                .AddSingleton<IHttpDownloadServiceFactory>(sp => sp.GetRequiredService<HttpDownloadServiceFactory>())
```

Comment:
Again no need to use an interface but just as an FYI the registration can be simplified, note that the `IVersionProvider`'s registration is different because the interface have limited methods but we always want to inject the same `VersionProvider` instance. This is not the case here:
```suggestion
                .AddSingleton<IHttpDownloadServiceFactory, HttpDownloadServiceFactory>()
```